### DISCUSSION TITLE
Extracting information about which ligand atoms form interactions with protein

### DISCUSSION THREAD

**Discussion by meddwl** [chemosim-lab/ProLIF#38] [Q&A]
ProLIF has such a wonderful opportunity as generating ligand-protein interactions fingerprint. If this fingerprint can be presented as a dataframe, it is possible to extract information about which residues interact with ligand. 
( There is an example how this dataframe looks like https://prolif.readthedocs.io/en/latest/source/modules/interaction-fingerprint.html )
Is there such an opportunity to expand this dataframe and extract information about which ligand atoms or a group of atoms form which type of interaction?

**Comment by cbouy** [comment]
Hi @meddwl,

Yes you can obtain the indices of atoms responsible for each interaction with `df = fp.to_dataframe(return_atoms=True)`.
The resulting dataframe now contains a pair of values instead of the usual `True` or `False` for each interaction.
These pair of values correspond to the indices of atoms in the ligand and protein residues.

For example, if you obtain the following dataframe:
```
ligand        LIG1.G      ...                                 
protein     ASP129.A      ...
interaction  HBDonor      ...
Frame                     ...                                 
0            (13, 8)      ...
```

This means that atom 13 in residue "LIG1.G" of your ligand, and atom 8 in residue "ASP129.A" of your protein are making that H-bond interaction in frame #0.

An quick example on how to use this:
```python
atom_idx = 13
resname = "LIG1.G"

lig = u.select_atoms("resname LIG")
lig_mol = plf.Molecule.from_mda(lig)
residue_mol = lig_mol[resname]
atom = residue_mol.GetAtomWithIdx(atom_idx)
coordinates = residue_mol.GetConformer().GetAtomPosition(atom_idx)
```

Also note that if an interaction wasn't present at a specific frame, you will have `(None, None)` instead of numbers.

Some examples of scripts to display which atoms are interacting directly on 2D or 3D structures are shown [here](https://prolif.readthedocs.io/en/latest/notebooks/visualisation.html).
